internal/subscriber/subscriber.go

```diff
@@ -571,7 +571,37 @@ func (s *Subscriber) pull(req *PullRequest) {
 				continue
 			}
 			ackID := ackMessage.BuildAckID()
-			sm = append(sm, &metrov1.ReceivedMessage{AckId: ackID, Message: protoMsg, DeliveryAttempt: msg.CurrentRetryCount + 1})
+			// check if the subscription has any filter applied and check if the message satisfies it if any
+
+			isElligible := true
```

Comment:
extract these to a separate method ?

Reply:
+1. Would suggest to write 1 method which accepts a message and returns bool if message should be delivered or not.

Reply:
Also, the eligible/not eligible check is being done once the message is added to the offset management heap. It should be done before that.

Reply:
nit: isEligible